src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/Hydrofracture.hpp

```diff
@@ -92,7 +92,7 @@ class Hydrofracture : public MGRStrategyBase< 1 >
     GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetPrintLevel( mgrData.coarseSolver.ptr, 0 ) );
     GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMaxIter( mgrData.coarseSolver.ptr, 1 ) );
     GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetTol( mgrData.coarseSolver.ptr, 0.0 ) );
-    GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMaxLevels( mgrData.coarseSolver.ptr, 25 ) );
+    GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMaxLevels( mgrData.coarseSolver.ptr, 1 ) );
```

Comment:
```suggestion
    GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMinCoarseSize( mgrData.coarseSolver.ptr, 1000 ) );
```

Could you try this options and check if you observe any improvement? If yes, we can further investigate whether `1000` is the appropriate minimum size. Thanks.

Reply:
@castelletto1 I tested this option, unfortunately, convergence issue persists for `pknViscosityDominated_benchmark.xml`
Below is the error message from running log:
```
++ Fracture propagation. Re-entering Newton Solve.
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
    ( R ) = ( 7.88e+00 ) ; 
{ 3, 1 }
		Linear Solver | Success | Iterations: 40 | Final Rel Res: 0.000473497 | Make Restrictor Time: 0 | Compute Auu Time: 0 | SC Filter Time: 0 | Setup Time: 3.15068 s | Solve Time: 7.87251 s
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
    ( R ) = ( 2.25e+12 ) ; 
    Last LinSolve(iter,res) = (  40, 4.73e-04 ) ; 
    The residual norm is above the maxAllowedResidualNorm of 1000000000. Newton loop terminated.
New dt = 5
    Attempt:  1, ConfigurationIter:  0, NewtonIter:  0
    ( R ) = ( 1.23e+09 ) ; 
    The residual norm is above the maxAllowedResidualNorm of 1000000000. Newton loop terminated.
New dt = 2.5
    Attempt:  2, ConfigurationIter:  0, NewtonIter:  0
    ( R ) = ( 6.13e+08 ) ; 
{ 3, 1 }
		Linear Solver | Success | Iterations: 26 | Final Rel Res: 0.000667621 | Make Restrictor Time: 0 | Compute Auu Time: 0 | SC Filter Time: 0 | Setup Time: 2.97576 s | Solve Time: 5.13145 s
    Attempt:  2, ConfigurationIter:  0, NewtonIter:  1
    ( R ) = ( 1.25e+04 ) ; 
    Last LinSolve(iter,res) = (  26, 6.68e-04 ) ; 
{ 3, 1 }
		Linear Solver | Success | Iterations: 47 | Final Rel Res: 0.000429517 | Make Restrictor Time: 0 | Compute Auu Time: 0 | SC Filter Time: 0 | Setup Time: 3.05405 s | Solve Time: 9.19932 s
    Attempt:  2, ConfigurationIter:  0, NewtonIter:  2
    ( R ) = ( 5.20e+03 ) ; 
    Last LinSolve(iter,res) = (  47, 4.30e-04 ) ; 
{ 3, 1 }
		Linear Solver | Success | Iterations: 54 | Final Rel Res: 0.000665758 | Make Restrictor Time: 0 | Compute Auu Time: 0 | SC Filter Time: 0 | Setup Time: 2.94388 s | Solve Time: 10.6859 s
    Attempt:  2, ConfigurationIter:  0, NewtonIter:  3
    ( R ) = ( 9.19e+13 ) ; 
    Last LinSolve(iter,res) = (  54, 6.66e-04 ) ; 
    The residual norm is above the maxAllowedResidualNorm of 1000000000. Newton loop terminated.
New dt = 1.25
    Attempt:  3, ConfigurationIter:  0, NewtonIter:  0
    ( R ) = ( 7.14e+10 ) ; 
    The residual norm is above the maxAllowedResidualNorm of 1000000000. Newton loop terminated.
New dt = 0.625
    Attempt:  4, ConfigurationIter:  0, NewtonIter:  0
    ( R ) = ( 3.57e+10 ) ; 
    The residual norm is above the maxAllowedResidualNorm of 1000000000. Newton loop terminated.
New dt = 0.3125
Convergence not achieved.
***** ERROR
***** LOCATION: /shared/scratch1/Users/j0551570/geosx/IntegratedTest/GEOS/src/coreComponents/physicsSolvers/SolverBase.cpp:763
***** Controlling expression (should be false): true
***** Rank 4: Nonconverged solutions not allowed. Terminating...
```

Reply:
@castelletto1 @CusiniM Should `maxAllowedResidualNorm` be boosted for nonlinear solver? as `maxAllowedResidualNorm="1e+15"`?

Reply:
Finally, increasing `maxAllowedResidualNorm` solves the problem

Reply:
Nice, thank you. What is linear solver option?

- `GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMaxLevels( mgrData.coarseSolver.ptr, 1 ) )`
- `GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMinCoarseSize( mgrData.coarseSolver.ptr, 1000 ) )`

Reply:
@castelletto1 `GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMinCoarseSize( mgrData.coarseSolver.ptr, 1000 ) )` is used

Reply:
Out of curiosity, can you check what happens if you revert back to setting the max number of levels to `1`? Thank you

Reply:
@castelletto1 The PKN case can also be successfully completed with option 1 (`GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetMaxLevels( mgrData.coarseSolver.ptr, 1 ) )`)